deploy-operator/Makefile

```diff
@@ -0,0 +1,80 @@
+
```

Comment:
I would prefer to avoid doing this in make if we can avoid it. Make is good for managing dependencies, but not for building ad-hoc commandline interfaces. In this case, since there are no dependency trees being managed by Make, I think this would be better as a shell script.

Reply:
Yeah this is one of the things that just came with kubebuilder, we don't really need `make`, it's just a couple of commands.

Reply:
If I remember our conversation from a while ago correctly, we only need the makefile to generate boilerplate from some kind of template, right? And the other targets in here are going to be replaced by `provisioner` blocks that emit Kubernetes YAML which we will `kubectl apply` into place. If that's true, I'd like to see this `Makefile` trimmed down to just what's needed for codegen, and then we should have a README in here explaining when codegen needs to be re-run and how to do it.

Reply:
The makefile is much shorter than what it was initially, but I'll remove some of the other targets too.

Reply:
We still need this makefile trimmed and some documentation of how the codegen works.